src/main/java/com/unigrad/funiverseauthenservice/security/services/ApplicationConfig.java

```diff
@@ -0,0 +1,51 @@
+package com.unigrad.funiverseauthenservice.security.services;
+
+
+import com.unigrad.funiverseauthenservice.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@RequiredArgsConstructor
+public class ApplicationConfig {
+
+    private final UserRepository repository;
+
+    @Bean
+    public UserDetailsService userDetailsService() {
+
+        return username -> repository.findByUsername(username)
```

Comment:
when updating to spring boot 3.x.x, it will implement like that to use UserDetailsService.
https://stackoverflow.com/questions/74672347/how-to-use-userdetailsservice-after-upgrade-spring-boot-2-7-x-to-spring-boot-3-0